libtsuba/src/LocalStorage.cpp

```diff
@@ -134,19 +134,17 @@ tsuba::LocalStorage::ListAsync(
   if ((dirp = opendir(dirname.c_str())) == nullptr) {
     if (errno == ENOENT) {
       // other storage backends are flat and so return an empty list here
-      return std::async(std::launch::deferred, []() -> katana::Result<void> {
-        return katana::ResultSuccess();
-      });
+      return std::async(
+          std::launch::deferred, []() -> AsyncError { return std::nullopt; });
```

Comment:
Didn't Arthur change a bunch of these to be `std::launch::deferred` just to avoid passing the thread-local result across threads?  Do we still want it?

Reply:
Yeah I think deferred is still the right thing for these very small no-work ops.